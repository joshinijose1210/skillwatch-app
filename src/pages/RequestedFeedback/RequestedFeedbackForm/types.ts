export type FeedbackTypeOption = 'positive' | 'improvement' | 'appreciation';
export type FeedbackTypeId = 1 | 2 | 3;
export type Action = 'Add' | 'View' | 'Edit';
export type Errors = Record<FeedbackTypeId, string>;

export interface FeedbackTypeObject {
    label: string;
    value: FeedbackTypeOption;
    feedbackTypeId: FeedbackTypeId;
}

export interface FeedbackItem {
    feedbackId?: number;
    feedbackTypeId: FeedbackTypeId;
    feedbackText: string;
    markdownText: string | null; // this should never be sent as empty string : ''
}

export interface FeedbackSectionProps {
    feedbackTagText: FeedbackTypeOption;
    feedbackTagNumber: FeedbackTypeId;
    action: Action;
}

export interface FeedbackFormContainerProps {
    handleAccordinToggle: (value: FeedbackTypeOption) => void;
    activeAccordion: FeedbackTypeOption | '';
}

export interface Tag {
    feedbackTypeId: FeedbackTypeId;
    feedbackType: FeedbackTypeOption;
}
export interface AccordionInputFieldProps {
    feedbackTypeOptions: FeedbackTypeObject[];
    handleAccordinToggle: (value: FeedbackTypeOption) => void;
    activeAccordion: FeedbackTypeOption | '';
    handleChange: (value: string, type: FeedbackTypeObject['feedbackTypeId']) => void;
    action: Action;
    feedbackErrors: Errors;
    getInputValue: (feedbackId: FeedbackTypeObject['feedbackTypeId']) => string;
}

// API response
export interface FeedbackDataItemResponse {
    feedbackId: number;
    feedback: string;
    feedbackTypeId: FeedbackTypeId;
    feedbackType: string;
    isDraft: boolean;
}
export interface SingleFeedbackResponse {
    organisationId: number;
    requestId: number;
    isSubmitted: boolean;
    isExternalRequest: boolean;
    requestedOn: string; // ISO date string
    submittedOn: string; // ISO date string
    actionItemId: number;
    actionItem: string;
    requestedById: number;
    requestedByEmployeeId: string;
    requestedByFirstName: string;
    requestedByLastName: string;
    feedbackToId: number;
    feedbackToEmployeeId: string;
    feedbackToFirstName: string;
    feedbackToLastName: string;
    feedbackFromId: number;
    feedbackFromEmployeeId: string;
    feedbackFromFirstName: string;
    feedbackFromLastName: string;
    externalFeedbackFromEmail: string;
    request: string;
    feedbackData: FeedbackDataItemResponse[];
}

export interface Payload {
    requestId: number;
    feedbackToId: number | string;
    feedbackFromId: number | string;
    organisationId: number | string;
    isDraft: boolean;
    feedbackData: {
        feedbackId?: number;
        feedbackTypeId: FeedbackTypeId;
        feedbackText: string;
        markdownText: string | null;
    }[];
}
export interface EditedDataPayload extends Omit<Payload, 'feedbackData'> {
    feedbackData: {
        feedbackId?: number;
        feedbackTypeId: FeedbackTypeId;
        feedbackText: string;
        markdownText: string | null;
        isNewlyAdded: boolean;
        isRemoved: boolean;
    }[];
}
