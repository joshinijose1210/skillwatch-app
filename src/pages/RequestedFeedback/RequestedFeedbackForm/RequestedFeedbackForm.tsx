import { ScreenWidthWrapper, StyledAccordionTitle, StyledTipsTitle } from '@common';
import { Loader, PageContent } from '@components';
import ListHeader from '@components/reusableComponents/ListHeader';
import TextDescription from '@components/reusableComponents/TextDescription';
import { Box, Button } from '@medly-components/core';
import { FC, memo } from 'react';
import {
    StyledButtonWrapper,
    FlexBoxRow,
    StyledAccordion,
    StyledAccordionHead,
    StyledAccordionContent,
    StyledHeading,
    StyledFeedbackFormTextArea,
    StyledText
} from './RequestedFeedbackForm.style';
import { useRequestedFeedbackForm } from './useRequestedFeedbackForm';
import * as Styled from '../../FeedbackForm/FeedbackForm.styled';
import GetDescription from '@pages/FeedbackForm/GetDescription';
import { TipHeader } from '@pages/FeedbackForm/FeedbackTipsDesc.styles';
import { AccordionInputFieldProps, FeedbackFormContainerProps, FeedbackSectionProps } from './types';
import { ThreeSixtyFeedbackRichTextEditor } from '@components/reusableComponents/RichTextEditor/360FeedbackRichtextEditor';

export const AccordionInputFields = memo(
    ({
        feedbackTypeOptions,
        handleAccordinToggle,
        activeAccordion,
        feedbackErrors,
        action,
        handleChange,
        getInputValue
    }: AccordionInputFieldProps) => {
        return (
            <>
                {feedbackTypeOptions.map((item, index) =>
                    //  in view feedback, show only the feedbacks that are submitted
                    action === 'View' ? (
                        getInputValue(item.feedbackTypeId).length > 0 && (
                            <StyledAccordion
                                key={index}
                                onChange={() => handleAccordinToggle(item.value)}
                                active={activeAccordion === item.value || false}
                            >
                                <StyledAccordionHead
                                    action={action}
                                    active={activeAccordion === item.value}
                                    hasError={feedbackErrors[item.feedbackTypeId].length > 0}
                                    feedbackTag={item.feedbackTypeId}
                                >
                                    <StyledAccordionTitle textWeight="Medium">{item.label}</StyledAccordionTitle>
                                </StyledAccordionHead>
                                <StyledAccordionContent active={activeAccordion === item.value}>
                                    <TextDescription
                                        disabled
                                        feedback={getInputValue(item.feedbackTypeId)}
                                        placeholder={'Feedback Shared'}
                                    />
                                </StyledAccordionContent>
                            </StyledAccordion>
                        )
                    ) : (
                        <StyledAccordion
                            key={index}
                            onChange={() => handleAccordinToggle(item.value)}
                            active={activeAccordion === item.value || false}
                        >
                            <StyledAccordionHead
                                action={action}
                                active={activeAccordion === item.value}
                                hasError={feedbackErrors[item.feedbackTypeId].length > 0}
                                feedbackTag={item.feedbackTypeId}
                            >
                                <TipHeader textWeight="Medium">{item.label}</TipHeader>
                            </StyledAccordionHead>
                            <StyledAccordionContent active={activeAccordion === item.value}>
                                <ThreeSixtyFeedbackRichTextEditor
                                    dataTestId={`${item.value}`}
                                    key={item.feedbackTypeId}
                                    useBlur={true}
                                    editor={'feedback'}
                                    input={getInputValue(item.feedbackTypeId)}
                                    marginBottom={1}
                                    // this is not for all, only if atleast one feedback is correctly added, rest become optional - handled in hook
                                    isOptional
                                    setInput={e => handleChange(e, item.feedbackTypeId)}
                                    placeholder="Write feedback here..."
                                />
                            </StyledAccordionContent>
                        </StyledAccordion>
                    )
                )}
            </>
        );
    }
);

AccordionInputFields.displayName = 'AccordionInputFields';

const FeedbackFormContainer = memo(({ handleAccordinToggle, activeAccordion }: FeedbackFormContainerProps) => {
    const {
        feedbackTypeOptions,
        isLoading,
        getInputValue,
        feedbackErrors,
        handleSend,
        buttonLoader,
        action,
        handleChange,
        isAddBtnDisabled
    } = useRequestedFeedbackForm();
    console.log(feedbackErrors, isAddBtnDisabled);
    return (
        <>
            <AccordionInputFields
                handleAccordinToggle={handleAccordinToggle}
                action={action}
                feedbackErrors={feedbackErrors}
                getInputValue={getInputValue}
                activeAccordion={activeAccordion}
                handleChange={handleChange}
                feedbackTypeOptions={feedbackTypeOptions}
            />
            {action !== 'View' && (
                <StyledButtonWrapper>
                    <Button
                        isLoading={buttonLoader === 1 && isLoading}
                        variant="outlined"
                        onClick={() => handleSend('Draft')}
                        disabled={isAddBtnDisabled}
                        data-testid="draftBtn"
                    >
                        Save as Draft
                    </Button>
                    <Button
                        data-testid="sendBtn"
                        disabled={isAddBtnDisabled}
                        onClick={() => handleSend('Save')}
                        isLoading={isLoading}
                        variant="solid"
                    >
                        Send
                    </Button>
                </StyledButtonWrapper>
            )}
        </>
    );
});

FeedbackFormContainer.displayName = 'FeedbackFormContainer';

export const FeedbackTipsSection = memo(({ feedbackTagNumber, feedbackTagText, action }: FeedbackSectionProps) => {
    return action === 'View' ? (
        <></>
    ) : (
        <ScreenWidthWrapper className="tips-division">
            <Styled.TipContainer feedbackTag={feedbackTagNumber}>
                <Styled.TipHeaderText>
                    <StyledTipsTitle>Tips</StyledTipsTitle>
                    <Styled.BulbImg />
                </Styled.TipHeaderText>
                <GetDescription feedbackTag={feedbackTagText} />
            </Styled.TipContainer>
        </ScreenWidthWrapper>
    );
});

FeedbackTipsSection.displayName = 'FeedbackTipsSection';

export const RequestedFeedbackForm: FC = () => {
    const {
        feedbackTagText,
        action,
        activeAccordion,
        handleAccordinToggle,
        feedbackAbout,
        feedbackDescription,
        feedbackFrom,
        actionItem,
        feedbackTagNumber,
        isLoading
    } = useRequestedFeedbackForm();

    return (
        <PageContent>
            <ListHeader
                title={
                    action == 'View' ? 'View Submitted Feedback' : action == 'Edit' ? 'Edit Requested Feedback' : 'Add Requested Feedback'
                }
            />
            <FlexBoxRow>
                <ScreenWidthWrapper className="form-division" action={action}>
                    {isLoading ? (
                        <Box style={{ alignItems: 'center', justifyContent: 'center', height: '30rem' }}>
                            <Loader />
                        </Box>
                    ) : (
                        <>
                            <Box style={{ flexDirection: 'column', padding: 0 }}>
                                <StyledHeading textVariant="h4" textWeight="Medium">
                                    Feedback Request From:
                                </StyledHeading>
                                <StyledText>{feedbackFrom}</StyledText>

                                <StyledHeading hasMarginTop textVariant="h4" textWeight="Medium">
                                    Feedback About:
                                </StyledHeading>
                                <StyledText>{feedbackAbout}</StyledText>

                                <StyledHeading hasMarginTop textVariant="h4" textWeight="Medium">
                                    Context:
                                </StyledHeading>
                                <StyledFeedbackFormTextArea
                                    feedbackData={feedbackDescription}
                                    dangerouslySetInnerHTML={{ __html: feedbackDescription }}
                                />
                                {actionItem && (
                                    <>
                                        <StyledHeading textVariant="h4" textWeight="Medium">
                                            Goal:
                                        </StyledHeading>
                                        <StyledFeedbackFormTextArea
                                            feedbackData={actionItem}
                                            dangerouslySetInnerHTML={{ __html: actionItem }}
                                        />
                                    </>
                                )}
                            </Box>
                            <FeedbackFormContainer activeAccordion={activeAccordion} handleAccordinToggle={handleAccordinToggle} />
                        </>
                    )}
                </ScreenWidthWrapper>
                <FeedbackTipsSection feedbackTagText={feedbackTagText} action={action} feedbackTagNumber={feedbackTagNumber} />
            </FlexBoxRow>
        </PageContent>
    );
};
