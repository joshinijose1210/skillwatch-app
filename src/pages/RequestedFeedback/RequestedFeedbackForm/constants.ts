import { colors } from '@pages/FeedbackForm/FeedbackForm.styled';

export const accordionHeaderBackgroundColors = colors;

export const accordionHeaderTitles = {
    1: 'Positive Feedback',
    2: 'Improvement Pointers',
    3: 'Appreciation Note'
};
