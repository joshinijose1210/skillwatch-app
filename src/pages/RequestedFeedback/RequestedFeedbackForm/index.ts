export { RequestedFeedbackForm as default } from './RequestedFeedbackForm';
