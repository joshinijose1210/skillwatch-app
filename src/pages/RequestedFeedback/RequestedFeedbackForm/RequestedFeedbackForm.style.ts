import { Accordion, Text } from '@medly-components/core';
import styled from 'styled-components';
import { accordionHeaderBackgroundColors } from './constants';
import { StyleTextArea } from '@components/reusableComponents/TextDescription/TextDescription.styled';
import { StyledProps } from '@common';

export const StyledButtonWrapper = styled('div')`
    display: flex;
    justify-content: flex-end;
    gap: 1rem;
`;

export const FlexBoxRow = styled('div')`
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    flex-direction: row;
`;

const getBorderColor = (hasError: boolean, action: 'Add' | 'View' | 'Edit', active?: boolean) => {
    if (active) {
        if (action === 'View') {
            return ' #98A7B7';
        } else if (action === 'Add' || action === 'Edit') {
            return '#c7d0d8';
        }
    } else if (hasError) return '#f03';
    else return 'transparent';
};

export const StyledAccordion = styled(Accordion)<StyledProps>``;
export const StyledAccordionHead = styled(Accordion.Header)<
    {
        hasError: boolean;
        feedbackTag: number;
        active?: boolean;
        action: 'Add' | 'View' | 'Edit';
    } & StyledProps
>`
    background-color: ${({ feedbackTag }) => accordionHeaderBackgroundColors[feedbackTag]};
    border: 1px solid ${({ hasError, active, action }) => getBorderColor(hasError, action, active)};
    border-bottom: 1px solid ${({ hasError, active, action }) => getBorderColor(hasError, action, active)};
    padding: 0.5rem 1.8rem;
`;

export const StyledAccordionContent = styled(Accordion.Content)<{ active?: boolean }>`
    margin-bottom: ${({ active }) => (active ? '1rem' : 0)};
`;

export const StyledHeading = styled(Text)<{ hasMarginTop?: boolean }>`
    margin-top: ${({ hasMarginTop }) => (hasMarginTop ? '1rem' : 0)};
    font-size: 1.4rem;
`;

export const StyledFeedbackFormTextArea = styled(StyleTextArea)`
    margin-top: unset;
    padding: unset;
    p {
        font-size: ${({ theme }) => theme.contentFontSize};
        margin: unset;
    }
`;

export const StyledText = styled(Text)`
    font-size: 1.5rem;
    font-weight: 400;
    line-height: 1.9rem;
    font-family: Open Sans, Helvetica Neue, Helvetica, Arial, sans-serif;
`;
