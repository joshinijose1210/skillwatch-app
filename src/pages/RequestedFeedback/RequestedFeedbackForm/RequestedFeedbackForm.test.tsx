import { RTKRender, fireEvent, screen, waitFor } from '@utils/test-utils';
import { RequestedFeedbackForm } from './RequestedFeedbackForm';
import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import { ManagerReviewMockData } from '@mocks/managerReview';
import { requestFeedbackMockData } from '@mocks/requestFeedback';
import * as reactRouterDom from 'react-router-dom';
import { feedbackFormMockData } from '@mocks/feedbackForm';

const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
    useLocation: jest.fn()
}));

beforeAll(() => {
    const { data } = requestFeedbackMockData.employee;
    mock.onGet(
        'https://dummy-api.example.com/api/employees/?teamId=-99&designationId=-99&organisationId=1&roleId=-99&sortOrderId=1&departmentId=-99'
    ).reply(200, data);
    const { url: feedbackTypeUrl, data: feedbackTypeData } = feedbackFormMockData.feedbackType;
    mock.onGet(feedbackTypeUrl).reply(200, feedbackTypeData);
});

const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');

describe('Requested FeedbackForm feedback form component for Add action', () => {
    beforeEach(() => {
        useLocationMock.mockReturnValue({
            ...ManagerReviewMockData.LocationMockData,
            state: { ...feedbackFormMockData.mockState, action: 'Add' }
        });
    });

    it('should take snapshot', () => {
        const { container } = RTKRender(<RequestedFeedbackForm />);
        expect(container).toMatchSnapshot();
    });

    it('should render propely', () => {
        RTKRender(<RequestedFeedbackForm />);
        expect(screen.getByText('Add Requested Feedback')).toBeInTheDocument();
        const SentBtn = screen.getByTestId('sendBtn');
        expect(SentBtn).toBeInTheDocument();
        expect(SentBtn).toBeDisabled();
    });

    it('should add request if correct data is entered', async () => {
        const { url: actionItemUrl, data: actionItemData } = feedbackFormMockData.actionItems;
        mock.onGet(actionItemUrl).replyOnce(200, actionItemData);
        mock.onPost('https://dummy-api.example.com/api/feedback/').replyOnce(200, {});
        const { container, rerender } = RTKRender(<RequestedFeedbackForm />);
        rerender(<RequestedFeedbackForm />);
        fireEvent.click(screen.getByText(/Appreciation Note/i));
        const editor = screen.getByTestId('appreciation');
        const htmlTag = container.querySelector('.ql-editor');
        const list = container.querySelector('.ql-list');
        fireEvent.focus(editor);
        fireEvent.blur(editor);
        list && fireEvent.click(list);
        if (htmlTag) {
            fireEvent.click(htmlTag);
            fireEvent.change(htmlTag, {
                target: {
                    innerHTML:
                        '<ul><li><strong><s>This a test feedback request just to verify the functionality of the component.</s></strong></li></ul>'
                }
            });
        }
        fireEvent.focus(editor);
        fireEvent.blur(editor);
        const SentBtn = screen.getByTestId('sendBtn');
        await waitFor(() => expect(SentBtn).toBeEnabled(), { timeout: 3000 });
        fireEvent.click(SentBtn);
        await screen.findByText('Requested Feedback submitted successfully');
    });

    it('should not add request if api fails', async () => {
        const { url: actionItemUrl, data: actionItemData } = feedbackFormMockData.actionItems;
        mock.onGet(actionItemUrl).replyOnce(200, actionItemData);
        mock.onPost('https://dummy-api.example.com/api/feedback/').replyOnce(400, {});
        const { container, rerender } = RTKRender(<RequestedFeedbackForm />);
        rerender(<RequestedFeedbackForm />);
        fireEvent.click(screen.getByText(/Improvement Pointers/i));
        const editor = screen.getByTestId('improvement');
        const htmlTag = container.querySelector('.ql-editor')?.getElementsByTagName('p');
        fireEvent.focus(editor);
        fireEvent.blur(editor);
        if (htmlTag !== undefined) {
            fireEvent.click(htmlTag[0]);
            fireEvent.change(htmlTag[0], {
                target: { innerHTML: 'This a test feedback request just to verify the functionality of the component.' }
            });
        }
        fireEvent.focus(editor);
        fireEvent.blur(editor);
        const SentBtn = screen.getByTestId('sendBtn');
        await waitFor(() => expect(SentBtn).toBeEnabled(), { timeout: 3000 });
        fireEvent.click(SentBtn);
        await screen.findByText('Sorry, Feedback not submitted. Please try again Later.');
    });
});

describe('Requested FeedbackForm feedback form component for Edit action', () => {
    beforeEach(() => {
        useLocationMock.mockReturnValue({
            ...ManagerReviewMockData.LocationMockData,
            state: { ...feedbackFormMockData.mockState, action: 'Edit' }
        });
    });

    it('should render propely', async () => {
        const { url, data } = requestFeedbackMockData.singleRequestedFeedback;
        mock.onGet(url).reply(200, data);
        RTKRender(<RequestedFeedbackForm />);
        expect(screen.getByText('Edit Requested Feedback')).toBeInTheDocument();
        await waitFor(() => expect(screen.queryByText('M A (SR0067)')), { timeout: 3000 });
        await waitFor(() => expect(screen.queryByText('R A (SR0068)')), { timeout: 3000 });
        expect(screen.getByText('Test request')).toBeInTheDocument();
        const SentBtn = screen.getByTestId('sendBtn');
        expect(SentBtn).toBeInTheDocument();
        expect(SentBtn).toBeEnabled();
    });

    it('should edit request and save as draft flow work properly', async () => {
        const { url: actionItemUrl, data: actionItemData } = feedbackFormMockData.actionItems;
        mock.onGet(actionItemUrl).replyOnce(200, actionItemData);
        mock.onPut('https://dummy-api.example.com/api/feedback/409').replyOnce(200, {});
        const { container } = RTKRender(<RequestedFeedbackForm />);
        fireEvent.click(screen.getByText(/Improvement Pointers/i));
        const editor = container.querySelector('[data-placeholder="Write feedback here..."]')?.getElementsByTagName('p');
        if (editor !== undefined) {
            fireEvent.change(editor[0], {
                target: { innerHTML: 'This a test feedback request just to verify the functionality of the component.' }
            });
        }
        const SentBtn = screen.getByTestId('draftBtn');
        await waitFor(() => expect(SentBtn).toBeEnabled(), { timeout: 3000 });
        fireEvent.click(SentBtn);
        await screen.findByText('Requested feedback saved successfully');
    });
});
