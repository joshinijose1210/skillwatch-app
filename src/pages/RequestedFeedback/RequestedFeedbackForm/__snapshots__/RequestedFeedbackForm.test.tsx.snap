// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Requested FeedbackForm feedback form component for Add action should take snapshot 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Add Requested Feedback
        </h3>
      </div>
    </div>
    <div
      class="sc-1sijxax-1 fmPQUh"
    >
      <div
        action="Add"
        class="sc-tkiih6-9 bFDgcw form-division"
      >
        <div
          bg="white"
          borderradius="2"
          class="sc-12ecrgm-0 dAEUnm"
          color="black"
          display="flex"
          font-variant="body2"
          px="4"
          py="4"
          style="align-items: center; justify-content: center; height: 30rem;"
        >
          <div
            class="sc-mbkfkf-0 KnoHL"
          >
            <main
              class="sc-1ehavxz-0 dDyhBJ"
            >
              <div
                class="sc-6d0kjn-0 dkGZbP"
              >
                <svg
                  class="sc-433e5x-0 dVjAFq"
                  height="1em"
                  viewBox="0 0 100 100"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M31.6 3.5C5.9 13.6-6.6 42.7 3.5 68.4c10.1 25.7 39.2 38.3 64.9 28.1l-3.1-7.9C44 97 19.9 86.6 11.5 65.3c-8.4-21.3 2-45.4 23.3-53.8l-3.2-8z"
                  >
                    <animatetransform
                      attributeName="transform"
                      attributeType="XML"
                      dur="2s"
                      from="0 50 50"
                      repeatCount="indefinite"
                      to="360 50 50"
                      type="rotate"
                    />
                  </path>
                  <path
                    d="M42.3 39.6c5.7-4.3 13.9-3.1 18.1 2.7 4.3 5.7 3.1 13.9-2.7 18.1l4.1 5.5c8.8-6.5 10.6-19 4.1-27.7-6.5-8.8-19-10.6-27.7-4.1l4.1 5.5z"
                  >
                    <animatetransform
                      attributeName="transform"
                      attributeType="XML"
                      dur="1s"
                      from="0 50 50"
                      repeatCount="indefinite"
                      to="-360 50 50"
                      type="rotate"
                    />
                  </path>
                  <path
                    d="M82 35.7C74.1 18 53.4 10.1 35.7 18S10.1 46.6 18 64.3l7.6-3.4c-6-13.5 0-29.3 13.5-35.3s29.3 0 35.3 13.5l7.6-3.4z"
                  >
                    <animatetransform
                      attributeName="transform"
                      attributeType="XML"
                      dur="2s"
                      from="0 50 50"
                      repeatCount="indefinite"
                      to="360 50 50"
                      type="rotate"
                    />
                  </path>
                </svg>
              </div>
            </main>
          </div>
        </div>
      </div>
      <div
        class="sc-tkiih6-9 bFDgcw tips-division"
      >
        <div
          class="sc-murh6i-6 gorXly"
          feedbacktag="1"
        >
          <div
            class="sc-murh6i-7 fggvmg"
          >
            <h3
              class="sc-1ntgl3l-0 lhJlYH sc-tkiih6-22 isRtxH"
              textalign="initial"
              textcolor="inherit"
              textvariant="h3"
              textweight="Strong"
            >
              Tips
            </h3>
            <div
              class="sc-murh6i-5 gpsljX"
            />
          </div>
          <strong
            class="sc-1ntgl3l-0 dDvaST sc-1xeus4s-0 hbEGSB"
            textalign="initial"
            textcolor="inherit"
            textweight="Strong"
          >
            <span>
              Spot the Good Stuff!
            </span>
          </strong>
          <br />
          <strong
            class="sc-1ntgl3l-0 dDvaST sc-1xeus4s-4 eGpFMS"
            margintop="2"
            textalign="initial"
            textcolor="inherit"
            textweight="Strong"
          >
            Purpose:
          </strong>
          <br />
          <span
            class="sc-1ntgl3l-0 hGagSU sc-1xeus4s-4 dWfSgT"
            textalign="initial"
            textcolor="inherit"
          >
            Reinforce behaviours that go above and beyond expectations.
          </span>
          <br />
          <strong
            class="sc-1ntgl3l-0 dDvaST sc-1xeus4s-4 eGpFMS"
            margintop="2"
            textalign="initial"
            textcolor="inherit"
            textweight="Strong"
          >
            What to do:
          </strong>
          <br />
          <ul
            class="sc-1xeus4s-1 hIctrP tips-steps"
          >
            <li
              class="sc-1xeus4s-3 iVvoKn"
              style="margin-top: 1rem;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1xeus4s-4 dWfSgT"
                textalign="initial"
                textcolor="inherit"
              >
                <span>
                  Be specific: Mention what the person did and how it helped.
                </span>
              </span>
            </li>
            <li
              class="sc-1xeus4s-3 iVvoKn"
              style="margin-top: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1xeus4s-4 dWfSgT"
                textalign="initial"
                textcolor="inherit"
              >
                <span>
                  Recognize above-and-beyond efforts, not just tasks that are part of the job.
                </span>
              </span>
            </li>
            <li
              class="sc-1xeus4s-3 iVvoKn"
              style="margin-top: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1xeus4s-4 dWfSgT"
                textalign="initial"
                textcolor="inherit"
              >
                <span>
                  Focus on behaviors and impact, not just outcomes.
                </span>
              </span>
            </li>
            <li
              class="sc-1xeus4s-3 iVvoKn"
              style="margin-top: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1xeus4s-4 dWfSgT"
                textalign="initial"
                textcolor="inherit"
              >
                <span>
                  Time it right: Give feedback close to when the action occurred.
                </span>
              </span>
            </li>
          </ul>
          <strong
            class="sc-1ntgl3l-0 dDvaST sc-1xeus4s-4 eGpFMS"
            margintop="2"
            textalign="initial"
            textcolor="inherit"
            textweight="Strong"
          >
            What not to do:
          </strong>
          <br />
          <ul
            class="sc-1xeus4s-1 hIctrP tips-steps"
          >
            <li
              class="sc-1xeus4s-3 iVvoKn"
              style="margin-top: 1rem;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1xeus4s-4 dWfSgT"
                textalign="initial"
                textcolor="inherit"
              >
                <span>
                  Giving praise for “just doing the job” — Example: “You attended all the meetings this week.” That's expected, not praise-worthy.
                </span>
              </span>
            </li>
            <li
              class="sc-1xeus4s-3 iVvoKn"
              style="margin-top: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1xeus4s-4 dWfSgT"
                textalign="initial"
                textcolor="inherit"
              >
                <span>
                  Generic compliments — Avoid “Great work!” without context.
                </span>
              </span>
            </li>
            <li
              class="sc-1xeus4s-3 iVvoKn"
              style="margin-top: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1xeus4s-4 dWfSgT"
                textalign="initial"
                textcolor="inherit"
              >
                <span>
                  Not acknowledging silent contributors — People who help behind the scenes also deserve appreciation.
                </span>
              </span>
            </li>
          </ul>
          <strong
            class="sc-1ntgl3l-0 dDvaST sc-1xeus4s-4 eGpFMS"
            margintop="2"
            textalign="initial"
            textcolor="inherit"
            textweight="Strong"
          >
            Example:
          </strong>
          <br />
          <span
            class="sc-1ntgl3l-0 hGagSU"
            style="font-size: 14px;"
            textalign="initial"
            textcolor="inherit"
          >
            "Loved how you took the initiative to redesign the onboarding doc. It made the process smoother for the new hires and saved the HR team hours!"
          </span>
        </div>
      </div>
    </div>
  </main>
</div>
`;
